### DISCUSSION TITLE
Does the order of functions in Handler.t.sol matter?

### DISCUSSION THREAD

**Discussion by aminsammara** [Cyfrin/foundry-full-course-cu#1866] [Q&A] [answered]
My question is regarding the order of functions defined in `Handler.t.sol`. 

If I define my Handler contract as follows:

```solidity
pragma solidity ^0.8.19;

contract Handler is Test {

function redeemCollateral() {..}
function depositCollateral() {..}

}
```

And have the `fail_on_revert=true` in my `foundry.toml` file, I still don't get any errors. But I'm just thinking about all the runs where `redeemCollateral()` is called first (without any prior collateral deposited) and how that should throw errors/reverts but it doesn't. 

What's going on here?

**Comment by EngrPips** [comment, marked as answer]
Hello @aminsammara, It wouldn't throw an error if the fuzzer call `redeemCollateral()`  before `depositCollateral()` because of the logic inside of `redeemCollateral()`, Let's look at what is the logic inside of `redeemCollateral()` and understand it.

** `redeemCollateral()` logic **

```js
function redeem_collateral(bool _pick_weth, uint256 _amount_of_collateral) public {
        // get the collateral for which we want to reedem
        ERC20Mock collateral = randomly_determine_token(_pick_weth);
        // set the maximum amount of the collateral we are going to be redeeming. Basically we make the maximum amount we can reedem be the amount the user have deposited in the dsc_engine initially 
        uint256 max_amount_of_collateral = dsc_engine.get_collateral_balance_of_user(address(collateral), msg.sender);

        // we bound the amount we will reedem to between the maximum amount that we can reedeem and 0
        _amount_of_collateral = bound(_amount_of_collateral, 0, max_amount_of_collateral);

        // if the amount `_amount_of_collateral` turned out to be `0` meaning the user have not deposited any amount for the particular collateral initially then we return which would not throw any error but just end the process
        if(_amount_of_collateral == 0) return;

        vm.prank(msg.sender);
        dsc_engine.redeemCollateral(address(collateral), _amount_of_collateral);
    }
```
Read through the comment in the code above and let me know if there is still any confusion.

**Reply by aminsammara** [reply]
It's so obvious when you break it down like this. Thank you :)

**Reply by EngrPips** [reply]
Awesome. I am glad you figured it out. Keep crushing it.
